voltixApp/VoltixApp/Chains/Sui.swift

```diff
@@ -0,0 +1,143 @@
+                $0.amounts = [UInt64(keysignPayload.toAmount)]
+            }
+            // 0.003 SUI
+            $0.signer = keysignPayload.coin.address
+            $0.gasBudget = 3000000
+            $0.referenceGasPrice = UInt64(referenceGasPrice)
+        }
+        
+        print(input.debugDescription)
+        
+        do {
+            let inputData = try input.serializedData()
+            return .success(inputData)
+        } catch {
+            return .failure(HelperError.runtimeError("fail to get PreSign input data: \(error.localizedDescription)"))
+        }
+    }
+    
+    
+    static func getPreSignedImageHash(keysignPayload: KeysignPayload) -> Result<[String], Error> {
+        let result = getPreSignedInputData(keysignPayload: keysignPayload)
+        switch result {
+        case .success(let inputData):
+            do {
+                let hashes = TransactionCompiler.preImageHashes(coinType: .sui, txInputData: inputData)
+                let preSigningOutput = try TxCompilerPreSigningOutput(serializedData: hashes)
+                print("ERROR preSigningOutput SUI \(preSigningOutput.errorMessage)")
+                return .success([Hash.blake2b(data: preSigningOutput.data, size: 32).hexString])
+            } catch {
+                return .failure(HelperError.runtimeError("fail to get preSignedImageHash,error:\(error.localizedDescription)"))
+            }
+        case .failure(let err):
+            return .failure(err)
+        }
+    }
+    
+    static func getSignedTransaction(vaultHexPubKey: String,
+                                     vaultHexChainCode: String,
+                                     keysignPayload: KeysignPayload,
+                                     signatures: [String: TssKeysignResponse]) -> Result<SignedTransactionResult, Error>
+    {
+        guard let pubkeyData = Data(hexString: vaultHexPubKey) else {
+            return .failure(HelperError.runtimeError("public key \(vaultHexPubKey) is invalid"))
+        }
+        guard let publicKey = PublicKey(data: pubkeyData, type: .ed25519) else {
+            return .failure(HelperError.runtimeError("public key \(vaultHexPubKey) is invalid"))
+        }
+        
+        let result = getPreSignedInputData(keysignPayload: keysignPayload)
+        switch result {
+        case .success(let inputData):
+            do {
+                let hashes = TransactionCompiler.preImageHashes(coinType: .sui, txInputData: inputData)
+                let preSigningOutput = try TxCompilerPreSigningOutput(serializedData: hashes)
+                print("ERROR preSigningOutput SUI \(preSigningOutput.errorMessage)")
+                let allSignatures = DataVector()
+                let publicKeys = DataVector()
+                let signatureProvider = SignatureProvider(signatures: signatures)
+                let signature = signatureProvider.getSignature(preHash: Hash.blake2b(data: preSigningOutput.data, size: 32))
```

Comment:
`Hash.blake2b(data: preSigningOutput.data, size: 32)` has been called twice , next to each other , can you please use a variable, and only do it once?